File: Android/TUIKit/TUICallKit/tuicallkit/src/main/java/com/tencent/qcloud/tuikit/tuicallkit/TUICallKit.java
package com.tencent.qcloud.tuikit.tuicallkit;

import android.content.Context;

import com.tencent.qcloud.tuikit.TUICommonDefine;
import com.tencent.qcloud.tuikit.tuicallengine.TUICallDefine;

import java.util.List;

public abstract class TUICallKit {

    public static TUICallKit createInstance(Context context) {
        return TUICallKitImpl.createInstance(context);
    }

    /**
     * Set user profile
     *
     * @param nickname User name, which can contain up to 500 bytes
     * @param avatar   User profile photo URL, which can contain up to 500 bytes
     *                 For example: https://liteav.sdk.qcloud.com/app/res/picture/voiceroom/avatar/user_avatar1.png
     * @param callback Set the result callback
     */
    public void setSelfInfo(String nickname, String avatar, TUICommonDefine.Callback callback) {
    }

    /**
     * Make a call
     *
     * @param userId        callees
     * @param callMediaType Call type
     */
    public void call(String userId, TUICallDefine.MediaType callMediaType) {
    }

    /**
     * Make a call
     *
     * @param userId        callees
     * @param callMediaType Call type
     * @param params        Extension param: eg: offlinePushInfo
     */
    public void call(String userId, TUICallDefine.MediaType callMediaType,
                     TUICallDefine.CallParams params, TUICommonDefine.Callback callback) {
    }

    /**
     * Make a group call
     *
     * @param groupId       GroupId
     * @param userIdList    List of userId
     * @param callMediaType Call type
     */
    public void groupCall(String groupId, List<String> userIdList, TUICallDefine.MediaType callMediaType) {
    }

    /**
     * Make a group call
     *
     * @param groupId       GroupId
     * @param userIdList    List of userId
     * @param callMediaType Call type
     * @param params        Extension param: eg: offlinePushInfo
     */
    public void groupCall(String groupId, List<String> userIdList,
                          TUICallDefine.MediaType callMediaType, TUICallDefine.CallParams params,
                          TUICommonDefine.Callback callback) {
    }

    /**
     * Join a current call
     *
     * @param roomId        current call room ID
     * @param callMediaType call type
     */
    public void joinInGroupCall(TUICommonDefine.RoomId roomId, String groupId, TUICallDefine.MediaType callMediaType) {
    }

    /**
     * Set the ringtone (preferably shorter than 30s)
     *
     * @param filePath Callee ringtone path
     */
    public void setCallingBell(String filePath) {
    }

    /**
     * Enable the mute mode (the callee doesn't ring)
     */
    public void enableMuteMode(boolean enable) {
    }

    /**
     * Enable the floating window
     */
    public void enableFloatWindow(boolean enable) {
    }
}

File: Android/TUIKit/TUICallKit/tuicallkit-kt/src/main/java/com/tencent/qcloud/tuikit/tuicallkit/TUICallKit.kt
package com.tencent.qcloud.tuikit.tuicallkit

import android.content.Context
import com.tencent.cloud.tuikit.engine.call.TUICallDefine
import com.tencent.cloud.tuikit.engine.call.TUICallDefine.CallParams
import com.tencent.cloud.tuikit.engine.common.TUICommonDefine
import com.tencent.cloud.tuikit.engine.common.TUICommonDefine.RoomId
import com.tencent.qcloud.tuikit.tuicallkit.common.data.Constants
import com.tencent.qcloud.tuikit.tuicallkit.view.CallAdapter

abstract class TUICallKit {
    companion object {
        @JvmStatic
        fun createInstance(context: Context): TUICallKit = TUICallKitImpl.createInstance(context)
    }

    /**
     * Set user profile
     *
     * @param nickname User name, which can contain up to 500 bytes
     * @param avatar   User profile photo URL, which can contain up to 500 bytes
     * For example: https://liteav.sdk.qcloud.com/app/res/picture/voiceroom/avatar/user_avatar1.png
     * @param callback Set the result callback
     */
    open fun setSelfInfo(nickname: String?, avatar: String?, callback: TUICommonDefine.Callback?) {}

    /**
     * Make a 1VN calls
     *
     * @param userIdList    List of userId
     * @param mediaType     Call type
     * @param params        Extension param: eg: offlinePushInfo
     */
    open fun calls(
        userIdList: List<String?>?, mediaType: TUICallDefine.MediaType,
        params: TUICallDefine.CallParams?, callback: TUICommonDefine.Callback?
    ) {
    }

    /**
     * Join a current call
     *
     * @param callId call Id
     */
    open fun join(callId: String?, callback: TUICommonDefine.Callback?) {}


    /**
     * Set the ringtone (preferably shorter than 30s)
     *
     * @param filePath Callee ringtone path
     */
    open fun setCallingBell(filePath: String?) {}

    /**
     * Enable the mute mode (the callee doesn't ring)
     */
    open fun enableMuteMode(enable: Boolean) {}

    /**
     * Enable the floating window
     */
    open fun enableFloatWindow(enable: Boolean) {}

    /**
     * Enable Virtual Background
     */
    open fun enableVirtualBackground(enable: Boolean) {}

    /**
     * Enable callee show banner view when received an new invitation
     * default: false
     */
    open fun enableIncomingBanner(enable: Boolean) {}

    /**
     * Set the display direction of the CallKit interface. The default value is portrait
     * @param orientation:  0-Portrait, 1-LandScape, 2-Auto;   default value: 0
     * Note: You are advised to use portrait mode to avoid abnormal display for small screen devices such as mobile phone
     */
    open fun setScreenOrientation(orientation: Int) {}

    /**
     * Disable the control button
     *
     * @param button value: microphone, audioDevice, camera, switchCamera, inviteUser.
     */
    open fun disableControlButton(button: Constants.ControlButton?) {}

    /**
     * set call adapter to customize call ui
     */
    open fun setAdapter(adapter: CallAdapter?) {}

    /**
     * Call experimental API
     *
     * @param jsonStr
     */
    open fun callExperimentalAPI(jsonStr: String) {}

    /**
     * Make a call
     *
     * @param userId        callees
     * @param callMediaType Call type
     */
    @Deprecated("Use NewInterface instead", ReplaceWith("calls"))
    open fun call(userId: String, callMediaType: TUICallDefine.MediaType) {
    }

    /**
     * Make a call
     *
     * @param userId        callees
     * @param callMediaType Call type
     * @param params        Extension param: eg: offlinePushInfo
     */
    @Deprecated("Use NewInterface instead", ReplaceWith("calls"))
    open fun call(
        userId: String, callMediaType: TUICallDefine.MediaType,
        params: CallParams?, callback: TUICommonDefine.Callback?
    ) {
    }

    /**
     * Make a group call
     *
     * @param groupId       GroupId
     * @param userIdList    List of userId
     * @param callMediaType Call type
     */
    @Deprecated("Use NewInterface instead", ReplaceWith("calls"))
    open fun groupCall(groupId: String, userIdList: List<String?>?, callMediaType: TUICallDefine.MediaType) {
    }

    /**
     * Make a group call
     *
     * @param groupId       GroupId
     * @param userIdList    List of userId
     * @param callMediaType Call type
     * @param params        Extension param: eg: offlinePushInfo
     */
    @Deprecated("Use NewInterface instead", ReplaceWith("calls"))
    open fun groupCall(
        groupId: String, userIdList: List<String?>?,
        callMediaType: TUICallDefine.MediaType, params: CallParams?,
        callback: TUICommonDefine.Callback?
    ) {
    }

    /**
     * Join a current call
     *
     * @param roomId        current call room ID
     * @param callMediaType call type
     */
    @Deprecated("Use NewInterface instead", ReplaceWith("join"))
    open fun joinInGroupCall(roomId: RoomId?, groupId: String?, callMediaType: TUICallDefine.MediaType?) {
    }

}